src/transpose.js

```diff
@@ -0,0 +1,42 @@
+var _curry1 = require('./internal/_curry1');
+
+
+/**
+ * Transposes the rows and columns of a 2D list.
+ * When passed a list of `n` lists of length `x`,
+ * returns a list of `x` lists of length `n`.
+ *
+ *
+ * @func
+ * @memberOf R
+ * @category List
+ * @sig transpose :: [[a]] -> [[a]]
+ * @param {Array} list A 2D list
+ * @return {Array} A 2D list
+ * @example
+ *
+ *      R.transpose([[1, 'a'], [2, 'b'], [3, 'c']]) //=> [[1, 2, 3], ['a', 'b', 'c']]
+ *      R.transpose([[1, 2, 3], ['a', 'b', 'c']]) //=> [[1, 'a'], [2, 'b'], [3, 'c']]
+ *
+ * If some of the rows are shorter than the following rows, their elements are skipped:
+ *
+ *      R.transpose([[10, 11], [20], [], [30, 31, 32]]) //=> [[10, 20, 30], [11, 31], [32]]
+ */
+module.exports = _curry1(function transpose(outerlist) {
+  var x = 0;
+  var result = [];
+  while (x < outerlist.length) {
+    var innerlist = outerlist[x];
+    var y = 0;
+    while (y < innerlist.length) {
+      if (!result.hasOwnProperty(y)) {
```

Comment:
I haven't quite wrapped my head around this implementation, but I wonder whether we could use a length check of some sort here. It seems odd to treat a list as a dictionary.